core/lib/zlib/zutil.h

```diff
@@ -187,6 +187,10 @@ extern z_const char * const z_errmsg[10]; /* indexed by 2-zlib_error */
 #endif
 
 /* provide prototypes for these when building zlib without LFS */
+#ifndef _LFS64_LARGEFILE
+#define _LFS64_LARGEFILE	0
```

Comment:
zlib changed quite a lot since 1.2.11 (from https://zlib.net/, the one we're based on) moved to zlib-ng (https://github.com/zlib-ng/zlib-ng) and its 2.0.x series. In the end, it seems zlib-ng expects `_LFS64_LARGEFILE` from _unistd.h_ but it seems to me `_LFS64_LARGEFILE` is not really a standard macro and may not be defined.

Should zlib be better compiled with `-D_LFS64_LARGEFILE=0`? Maybe.

Reply:
addressed (force pushed)